domain/consensus/timelock_CLTV_test.go

```diff
@@ -0,0 +1,308 @@
+		}
+		currentBlockHeight++
+		blockC, err := testConsensus.GetBlock(blockCHash)
+		if err != nil {
+			t.Fatalf("Failed getting blockC: %v", err)
+		}
+		fees := uint64(1)
+		fundingTransaction, err := testutils.CreateTransaction(blockC.Transactions[transactionhelper.CoinbaseTransactionIndex], fees)
+		if err != nil {
+			t.Fatalf("Error creating fundingTransaction: %v", err)
+		}
+		blockDHash, _, err := testConsensus.AddBlock([]*externalapi.DomainHash{blockCHash}, nil,
+			[]*externalapi.DomainTransaction{fundingTransaction})
+		if err != nil {
+			t.Fatalf("Error creating blockD: %v", err)
+		}
+		currentBlockHeight++
+		//Create a CLTV script:
+		targetBlockHeight := int64(30)
+		redeemScriptCLTV, err := createScriptCLTV(targetBlockHeight)
+		if err != nil {
+			t.Fatalf("Failed to create a script using createScriptCLTV: %v", err)
+		}
+		p2shScriptCLTV, err := txscript.PayToScriptHashScript(redeemScriptCLTV)
+		if err != nil {
+			t.Fatalf("Failed to create a pay-to-script-hash script : %v", err)
+		}
+		scriptPublicKeyCLTV := externalapi.ScriptPublicKey{
+			Version: constants.MaxScriptPublicKeyVersion,
+			Script:  p2shScriptCLTV,
+		}
+		transactionWithLockedOutput, err := createTransactionWithLockedOutput(fundingTransaction, fees, &scriptPublicKeyCLTV)
+		if err != nil {
+			t.Fatalf("Error in createTransactionWithLockedOutput: %v", err)
+		}
+		// BlockE contains the locked output (locked by CLTV).
+		// This block should be valid since CLTV script locked only the output.
+		blockEHash, _, err := testConsensus.AddBlock([]*externalapi.DomainHash{blockDHash}, nil,
+			[]*externalapi.DomainTransaction{transactionWithLockedOutput})
+		if err != nil {
+			t.Fatalf("Error creating blockE: %v", err)
+		}
+		currentBlockHeight++
+		// Create a transaction that tries to spend the locked output.
+		transactionThatSpentTheLockedOutput, err := createTransactionThatSpentTheLockedOutput(transactionWithLockedOutput,
+			fees, redeemScriptCLTV, targetBlockHeight)
+		if err != nil {
+			t.Fatalf("Error creating transactionThatSpentTheLockedOutput: %v", err)
+		}
+		// Add a block that contains a transaction that spends the locked output before the time, and therefore should be failed.
+		// (The block height should be x, before the output will be spendable, where x = 'targetBlockHeight' ).
+		_, _, err = testConsensus.AddBlock([]*externalapi.DomainHash{blockEHash}, nil,
+			[]*externalapi.DomainTransaction{transactionThatSpentTheLockedOutput})
+		if err == nil || !errors.Is(err, ruleerrors.ErrUnfinalizedTx) {
+			t.Fatalf("Expected block to be invalid with err: %v, instead found: %v", ruleerrors.ErrUnfinalizedTx, err)
+		}
+
+		// Add blocks to release the locked output, the block height should be 'numOfBlocksToWait'.
+		tipHash := blockEHash
+		numOfBlocksToAdd := targetBlockHeight - currentBlockHeight
+		for i := int64(0); i < numOfBlocksToAdd; i++ {
+			tipHash, _, err = testConsensus.AddBlock([]*externalapi.DomainHash{tipHash}, nil, nil)
```

Comment:
You can also stop when the DAA score of the tip equals to the target, instead of keeping track of currentBlockHeight